### PR TITLE
Imply-4878 Update build process to publish jar to local mvn for implydata/netty

### PR DESCRIPTION
**Pull request by linsong-imply**
Motivation:

Publish artifact into our local maven repo, so that it can be referenced in the Druid project.

Modification:

Added Jenkins file to build and publish artifact.

Result:

The artifact is published.
https://artifactory.qa.imply.io/artifactory/webapp/#/artifacts/browse/tree/General/imply-libs-release-local/io/netty/netty

Jenkins Job
https://ci.qa.imply.io/job/imply-druid/job/netty/job/IMPLY-4878/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                    sh "docker pull $ECR/$IMAGE"
+                }
+            }
+        }
+
+        stage ('Build and Publish') {
+            steps {
+                sh """
+                    echo "mvn versions:set -B -DremoveSnapshot | true" > run.sh
+                    echo "jfrog rt mvn clean install" >> run.sh
```
Comment (suneet-s) [inline Jenkinsfile:21]: What is the `run.sh` script?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                    docker run -v ${WORKSPACE}/.m2:/root/.m2 -v ${WORKSPACE}:/app $ECR/$IMAGE bash run.sh
+                """
+            }
```
Comment (suneet-s) [inline Jenkinsfile:37]: How is the snapshot version bumped after the artifact is built and published?

Comment (linsong-imply) [reply Jenkinsfile:37]: I thought you don't need snapshot versions.
But here is what I tested.
https://artifactory.qa.imply.io/artifactory/webapp/#/artifacts/browse/tree/General/imply-libs-snapshot-local/io/netty/netty/3.10.6.Final-iap3-SNAPSHOT
image

Comment (suneet-s) [reply Jenkinsfile:37]: We don't need SNAPSHOTS to be published, but can you describe the high level workflow of adding fixes to the netty repo, and how they get published with a new version and then how Druid will pick up the new version

Comment (linsong-imply) [reply Jenkinsfile:37]: I am updating the build process to use our new artifactory at https://implydata.jfrog.io/, will add readme asap.

Comment (linsong-imply) [reply Jenkinsfile:37]: The readme is updated

File: README.md [hunk 3]
```diff
@@ -1,3 +1,48 @@
...
+
+Update your projects to use Imply netty artifact stored in Imply Artifactory.
+
+### ~/.m2/settings.xml
+
+```
+    <servers>
+        <server>
+            <username>eng</username>
+            <password>AP8cigvgtfjXGXLig7xcFUCWqJG</password>
```
Comment (suneet-s) [inline README.md:11]: Is it ok to store this password in clear text in other repos?

Comment (suneet-s) [reply README.md:11]: oops dumb question - please ignore

**Review by suneet-s — APPROVED** [review]

**Comment by suneet-s** [comment]
Nice! 👍

**Pull request merged**